src/D2L.Bmx/Okta/SourceGenerationContext.cs

```diff
@@ -0,0 +1,14 @@
+using System.Text.Json.Serialization;
+using D2L.Bmx.Okta.Models;
+namespace D2L.Bmx.Okta;
+
+[JsonSourceGenerationOptions( PropertyNamingPolicy = JsonKnownNamingPolicy.CamelCase )]
+[JsonSerializable( typeof( AuthenticateOptions ) )]
+[JsonSerializable( typeof( AuthenticateChallengeMfaOptions ) )]
+[JsonSerializable( typeof( SessionOptions ) )]
+[JsonSerializable( typeof( AuthenticateResponseInital ) )]
+[JsonSerializable( typeof( AuthenticateResponseSuccess ) )]
+[JsonSerializable( typeof( OktaSession ) )]
+[JsonSerializable( typeof( OktaApp[] ) )]
+internal partial class SourceGenerationContext : JsonSerializerContext {
```

Comment:
Interesting, https://devblogs.microsoft.com/dotnet/try-the-new-system-text-json-source-generator/#why-do-source-generation 

I guess BMX is a good place to trial this sort of thing.